### DISCUSSION TITLE
Support html and pdf as input file type

### DISCUSSION THREAD

**Discussion by chenyang-shanghai** [microsoft/graphrag#375] [Q&A]
In the config guide,  we can only chose .csv or text for the "file_type";
https://microsoft.github.io/graphrag/posts/config/json_yaml/

Whereas html file (saved from internal web site) and pdf are very common now. So I'd ask whether the Microsoft graphrag supports these input files types or not; If it supports, any guide will be appreciated.

**Comment by cendenta** [comment]
Integration with [Document Intelligence and all its supported types](https://learn.microsoft.com/en-us/azure/ai-services/document-intelligence/concept-read?view=doc-intel-4.0.0&tabs=sample-code#input-requirements) would be excellent too.

**Reply by RichMorin** [reply]
HTML and PDF are plausible input formats, but they aren't the only ones of interest. To explore this, let's begin by comparing them with the current (CSV and "text") categories. Although CSV is written as a text (as opposed to binary) file, it is optimized for encoding rows of values (e.g., numbers, strings). Typically, this is used to describe a rectangular data structure (e.g., spreadsheet, table).

However, vanilla CSV does not include syntactic support for links, styles, etc. So, HTML and PDF would typically include information that CSV cannot represent without extensions. For example, it seems that Microsoft Excel defines a [=HYPERLINK()] (https://stackoverflow.com/questions/7572268/how-to-encode-a-hyperlink-in-csv-formatted-file) extension for links, but GR may not support this.

That said, in the world of LLMs and such, any sequence of characters can be treated as a token and thus serve as an implicit link between the usage instances. So, can someone point me to a description of what GraphRAG considers to be "text" and what sorts of graph information GR can extract from it?

Also, if GR can harvest linkage information from fairly arbitrary text files, perhaps it could make sense of some popular graph representation formats. For example:

- [ArangoDB](https://en.wikipedia.org/wiki/ArangoDB) is a graph database which is based on [JSON](https://en.wikipedia.org/wiki/JSON). 
- [Graphviz](https://en.wikipedia.org/wiki/Graphviz) uses the [DOT](https://en.wikipedia.org/wiki/DOT_(graph_description_language)) graph description language.
- [JSON-LD](https://en.wikipedia.org/wiki/JSON-LD) is a standard for encoding [linked data](https://en.wikipedia.org/wiki/Linked_data).
- [Neo4j](https://en.wikipedia.org/wiki/Neo4j) uses the [Cypher query language.](https://en.wikipedia.org/wiki/Cypher_(query_language)]).

Can someone fill me in on the current prospects for this sort of thing?

-r

**Comment by AlonsoGuevara** [comment]
Hi! Currently we only support txt and csv inputs, as you pointed out.

However, what I have done to process multiple file types is to have a preprocessing step where, using libs like pypdf, I convert everything into the same format. Same for docx, xslx, among others.

I really like the proposal of adding Document Intelligence support.

**Reply by adairgj** [reply]
What part of the architecture restricts it to txt and csv only?  The LLM supports these formats so I'm curious why its restricted.

**Reply by matbeedotcom** [reply]
> What part of the architecture restricts it to txt and csv only? The LLM supports these formats so I'm curious why its restricted.

it's probably just its token truncating / token chunking system, to be honest.

Text tends to be formatted with newlines, paragraphs, etc... I'll have to take a deeper look

**Comment by Bioinf-usr** [comment]
+1 for the document intelligence support. The new version of document intelligence is really good. Would be wonderful to have it integrated as part of graphRAG processing directly.

**Comment by dyafu** [comment]
use docling do an ocr on the pdf and then feed the text to graphrag
